scripts/clang-tidy.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+# https://clang.llvm.org/extra/clang-tidy/
+
+: '
+Runs clang-tidy on the code in src/
+Return `1` if there are files automatically fixed by clang-tidy.
+Returns `0` if no fixes by clang-tidy.
+TODO: should also return non-zero if clang-tidy emits warnings
+or errors about things it cannot automatically fix. However I cannot
+figure out how to get this working yet as it seems that clang-tidy
+always returns 0 even on errors.
+'
+
+PATH_TO_CLANG_TIDY_SCRIPT="$(pwd)/mason_packages/.link/share/run-clang-tidy.py"
+# make sure that run-clang-tidy.py can find the right clang-tidy
+export PATH=$(pwd)/mason_packages/.link/bin:${PATH}
+
+# build the compile_commands.json file if it does not exist
```

Comment:
Why? Why do we need the compile commands database?

Reply:
See above - needed by `run-clang-tidy.py`